app/uk/gov/hmrc/tai/model/hip/reads/NpsIabdSummaryHipReads.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tai.model.hip.reads
+
+import play.api.libs.json.*
+import uk.gov.hmrc.tai.model.domain.NpsIabdSummary
+import uk.gov.hmrc.tai.util.JsonHelper.readsTypeTuple
+
+object NpsIabdSummaryHipReads {
+
+  private val categories =
+    Seq("nonSavings", "untaxedInterest", "bankInterest", "ukDividends", "foreignInterest", "foreignDividends")
+  private val listNames = Seq("summaryIABDDetailsList", "summaryIABDEstimatedPayDetailsList")
+  private val subPaths = Seq("totalIncomeDetails", "allowanceReliefDeductionsDetails")
+
+  val iabdsFromTotalLiabilityReads: Reads[Seq[NpsIabdSummary]] =
+    Reads { json =>
+      JsSuccess {
+        categories.flatMap { category =>
```

Comment:
A for .. yield with a flatten on the end might be more concise and clearer here. Below too